src/modules/shared/input-converter.ts

```diff
@@ -1,31 +1,41 @@
-const stringConverter = (value: any) => {
+import {} from 'reflect-metadata';
+
+// Inspiration for this script was taken from:
+// https://blog.rsuter.com/
+//  angular-2-typescript-property-decorator-that-converts-input-values-to-the-correct-type/
+
+export const stringConverter = (value: any) => {
   if (value === undefined || typeof value === 'string') {
     return value;
   }
 
   return value.toString();
 };
 
-const booleanConverter = (value: any) => {
+export const booleanConverter = (value: any) => {
   if (value === undefined || typeof value === 'boolean') {
     return value;
   }
 
   return value.toString() === 'true';
 };
 
-const numberConverter = (value: any) => {
+export const numberConverter = (value: any) => {
   if (value === undefined || typeof value === 'number') {
     return value;
   }
 
   return parseFloat(value.toString());
 };
 
+const getMetaData = (target: Object, key: string) => {
+  return Reflect.getMetadata('design:type', target, key);
+};
+
 export function InputConverter(converter?: (value: any) => any) {
```

Comment:
Yeah, exporting functions makes them really difficult to spy on. We need to remember this for the future. We should probably make it a design pattern.